app/components/badge_component.rb

```diff
@@ -0,0 +1,18 @@
+class BadgeComponent < ViewComponent::Base
+  DARK_TEXT_TYPES = [:warning, :light]
+
+  attr_reader :text, :type, :rounded, :margin
```

Comment:
You don't technically need this line if you just use `@` notation in the style method.

Reply:
You're right. I guess only the `text` attribute needs to be accessible, the other ones can be private.